Dockerfile

```diff
@@ -0,0 +1,112 @@
+# Stage 1:
+# - base: ubuntu (default) OR prebuilt image0
+# - install build tools
+# - run playbook (image0 avoids rerunning lengthy tasks)
+# - remove build artifacts + files not needed in container
+# Stage 2:
+# - install python-virtualenv
+# - create galaxy user + group + directory
+# - copy galaxy files from stage 1
+# - finalize container (set path, user...)
+
+# Init ARGs
+ARG ROOT_DIR=/galaxy
+ARG SERVER_DIR=$ROOT_DIR/server
+# For much faster build time override this with image0 (Dockerfile.0 build):
+#   docker build --build-arg BASE=<image0 name>...
+ARG STAGE1_BASE=ubuntu:20.04
+ARG STAGE2_BASE=$STAGE1_BASE
+# NOTE: the value of GALAXY_USER must be also hardcoded in COPY in final stage
+ARG GALAXY_USER=galaxy
+ARG GALAXY_PLAYBOOK_REPO=https://github.com/galaxyproject/galaxy-docker-k8s
+
+# Stage-1
+FROM $STAGE1_BASE AS stage1
+ARG DEBIAN_FRONTEND=noninteractive
+ARG SERVER_DIR
+ARG GALAXY_PLAYBOOK_REPO
+
+# Install build dependencies + ansible
+RUN set -xe; \
+    apt-get -qq update && apt-get install -y --no-install-recommends \
+        apt-transport-https \
+        git \
+        make \
+        python3-virtualenv \
+        python3-dev \
+        software-properties-common \
+        ssh \
+        gcc \
+        libpython3.6 \
+        locales locales-all \
+    && apt-get -qq update && apt-get install -y --no-install-recommends \
+        ansible \
+    && apt-get autoremove -y && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* /tmp/*
+
+# Remove context from previous build; copy current context; run playbook
+WORKDIR /tmp/ansible
+RUN rm -rf *
+ENV LC_ALL en_US.UTF-8
+RUN git clone --depth 1 $GALAXY_PLAYBOOK_REPO galaxy-docker
+WORKDIR /tmp/ansible/galaxy-docker
+RUN ansible-galaxy install -r requirements.yml -p roles --force-with-deps
+
+# Add Galaxy source code
+COPY . $SERVER_DIR/
+RUN ansible-playbook -i localhost, playbook.yml -vv
+
+RUN cat /galaxy/server/lib/galaxy/dependencies/conditional-requirements.txt | grep psycopg2-binary | xargs /galaxy/server/.venv/bin/pip install
+
+# Remove build artifacts + files not needed in container
+WORKDIR $SERVER_DIR
+# Save commit hash of HEAD before zapping git folder
+RUN git rev-parse HEAD > GITREVISION
+RUN rm -rf \
+        .ci \
+        .git \
+        .venv/bin/node \
```

Comment:
You'll need node for expression tools.

Reply:
Hmm... That probably means that expression tools don't work in k8s at the moment.
Which of these folders do you think we can safely zap? `client/node_modules` is probably ok I'm guessing?

Reply:
@nuwang Sorry, I missed this. Yes, I *think* it is safe to drop `client/node_modules` - that's what we did back in 2019.

Reply:
I think expression tools are mapped to another container in k8s ... we did fix this last year for the covid workflows.

Reply:
Given this work needs that extra K8S context for expression tools I've renamed the Dockerfile to ".k8s_ci.Dockerfile" instead of ".ci.Dockerfile" as I suggested elsewhere.